core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -52,6 +52,7 @@ public final class DefaultEc2Metadata implements Ec2Metadata {
 
     private static final RequestMarshaller REQUEST_MARSHALLER = new RequestMarshaller();
 
+    private static final EndpointProvider ENDPOINT_PROVIDER = EndpointProvider.builder().build();
```

Comment:
Maybe just a TODO, but we should reuse the profile name and file from the client, when we're using this IMDS client as part of the IMDS credentials provider.